### PR TITLE
Added aten::split_with_sizes operation.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/csrc/tensor.cpp [hunk 1]
```diff
@@ -1206,23 +1212,33 @@ std::vector<XLATensor> XLATensor::split(const XLATensor& input,
...
                    input.GetDevice())};
   }
-  xla::int64 chunks = xla::CeilOfRatio(size_in_dim, split_size);
-  ir::NodePtr node =
-      ir::MakeNode<ir::ops::Split>(input.GetIrValue(), split_size, split_dim);
-  std::vector<XLATensor> result;
-  for (xla::int64 i = 0; i < chunks; ++i) {
-    result.push_back(Create(ir::Value(node, i), input.GetDevice()));
+  std::vector<xla::int64> split_sizes;
+  for (; dim_size > 0; dim_size -= split_size) {
```
Comment (asuhan) [inline torch_xla/csrc/tensor.cpp]: I'd use a `while` here, it's almost as concise and more readable.

File: torch_xla/csrc/data_ops.cpp [hunk 2]
```diff
@@ -276,14 +276,35 @@ std::vector<xla::XlaOp> BuildChunk(const torch::jit::Node* node,
...
-                                   xla::int64 split_size, xla::int64 dim) {
+size_t ComputeSplitCount(
+    xla::int64 dim_size,
+    tensorflow::gtl::ArraySlice<const xla::int64> split_sizes) {
+  size_t count = 0;
+  for (auto size : split_sizes) {
+    if (size > dim_size) {
+      break;
+    }
+    dim_size -= size;
```
Comment (asuhan) [inline torch_xla/csrc/data_ops.cpp]: Let's use a `remaining_dim_size` variable instead, reusing the input parameter is confusing.

File: torch_xla/csrc/tensor.cpp (hunk 1 above)
Comment (dlibenzi) [reply torch_xla/csrc/tensor.cpp]: Why?
I can understand "for (; COND;)" is silly.
I can even (barely) understand a "for(;;)" -> "while (true)" migration.
But "for (; COND; POST)" expanded to a while it is objectively more concise, and personally more readable ;)

File: torch_xla/csrc/data_ops.cpp (hunk 2 above)
Comment (dlibenzi) [reply torch_xla/csrc/data_ops.cpp]: Why? I could understand if the function was 100 lines long. But this fits in 1/10 of a screen, and clearly shows that the variable is not used elsewhere.

**Pull request merged**